addons/mail/static/src/messaging/entity/composer/composer.js

```diff
@@ -191,6 +237,43 @@ function ComposerFactory({ Entity }) {
 
         /**
          *
+         * Generates the html link related to the mentionned partner
+         *
+         * @private
+         * @param {string} body
+         * @returns {string}
+         */
+        _generateMentionsLinks(body) {
+            if (this.mentionedPartners.length === 0) {
+                return body;
+            }
+            const inputMentions = body.match(new RegExp("@"+'[^ ]+(?= |&nbsp;|$)', 'g'));
+            const substrings = [];
+            let startIndex = 0;
+            for (const i in inputMentions) {
+                const match = inputMentions[i];
+                const matchName = match.substring(1).replace(new RegExp('\u00a0', 'g'), ' ');
+                const endIndex = body.indexOf(match, startIndex) + match.length;
+                const partner = this.mentionedPartners.find(partner => partner.name === matchName);
+                let mentionLink = "@" + matchName;
+                if (partner) {
+                    const baseHREF = this.env.session.url('/web');
+                    const href = `href='${baseHREF}#model=res.partner&id=${partner.id}'`;
+                    const attClass = `class='o_mail_redirect'`;
+                    const dataOeId = `data-oe-id='${partner.id}'`;
+                    const dataOeModel = `data-oe-model='res.partner'`;
+                    const target = `target='_blank'`;
+                    mentionLink = `<a ${href} ${attClass} ${dataOeId} ${dataOeModel} ${target} >@${matchName}</a>`;
```

Comment:
If we keep it, we must ensure every part that is built manually is properly escaped to avoid injection.

Reply:
Sorry to ask but as it is, I can't see where there could be an injection

Reply:
The possible injection is because ultimately the template is doing a `t-raw` with this content.

It's much better to be safe and escape values preemptively, where the raw HTML is build, instead of relying on a potential post-processing.

The most potentially problematic part here is the "${matchName}" since this is related to user input.
You can wrap the name with https://github.com/odoo/owl/blob/master/doc/reference/utils.md#escape

Other "parts" of this HTML are supposedly safe since they are just strings that are hardcoded above or relying on `partner.id` which can decently be considered safe (though the best would be to escape it too just to be totally safe).

Reply:
Yes you can use either, if we have a native/owl solution it is better than _ which is another external lib.
There is probably no use of owl escape in the code yet since it's new.